ostd/src/mm/kspace/kva.rs

```diff
@@ -0,0 +1,287 @@
+                }
+            }
+        }
+
+        if let Some(range) = allocate_range {
+            Ok(Self { var: range })
+        } else {
+            Err(Error::KvaAllocError)
+        }
+    }
+
+    fn dealloc(&self, freelist: &mut BTreeMap<Vaddr, KvaFreeNode>, range: Range<Vaddr>) {
+        freelist.insert(range.start, KvaFreeNode::new(range.start..range.end));
+    }
+    // /// Un-map pages mapped in kernel virtual area.
+    // /// # Safety
+    // /// The caller should ensure that the operation doesn't violate the memory safety of
+    // /// kernel objects.
+    // unsafe fn unmap<T>(&mut self, range: Range<Vaddr>, pages: Vec<Page<T>>) {
+    //     //
+    //     todo!();
+    // }
+    /// Set the property of the mapping.
+    /// # Safety
+    /// The caller should ensure that the protection doesn't violate the memory safety of
+    /// kernel objects.
+    unsafe fn protect(&mut self, range: Range<Vaddr>, op: impl FnMut(&mut PageProperty)) {
+        assert!(range.start >= self.var.start && range.end <= self.var.end);
+        let page_table = KERNEL_PAGE_TABLE.get().unwrap();
+        let mut cursor = page_table.cursor_mut(&range).unwrap();
+        cursor.protect(range.len(), op, true).unwrap();
+    }
+
+    // Maybe other advanced object R/W methods like what's offered in the safe version?
+}
+
+impl<'a> KvaInner {
+    /// Returns a reader to read data from it.
+    pub fn reader(&'a self) -> VmReader<'a> {
+        unsafe { VmReader::from_kernel_space(self.var.start as *const u8, self.var.len()) }
+    }
+
+    /// Returns a writer to write data into it.
+    pub fn writer(&'a self) -> VmWriter<'a> {
+        unsafe { VmWriter::from_kernel_space(self.var.start as *mut u8, self.var.len()) }
+    }
+}
+
+impl VmIo for KvaInner {
+    // `VmIo` counterparts
+    fn read_bytes(&self, offset: usize, buf: &mut [u8]) -> Result<()> {
+        // Do bound check with potential integer overflow in mind
+        let max_offset = offset.checked_add(buf.len()).ok_or(Error::Overflow)?;
+        if max_offset > self.var.len() {
+            return Err(Error::InvalidArgs);
+        }
+        let len = self.reader().skip(offset).read(&mut buf.into());
+        debug_assert!(len == buf.len());
+        Ok(())
+    }
+
+    fn write_bytes(&self, offset: usize, buf: &[u8]) -> Result<()> {
+        // Do bound check with potential integer overflow in mind
+        let max_offset = offset.checked_add(buf.len()).ok_or(Error::Overflow)?;
+        if max_offset > self.var.len() {
+            return Err(Error::InvalidArgs);
+        }
+        let len = self.writer().skip(offset).write(&mut buf.into());
+        debug_assert!(len == buf.len());
+        Ok(())
+    }
+}
+
+// static KVA_FREELIST: SpinLock<BtreeMap<Vaddr, KvaFreeNode>> = SpinLock::new(BtreeMap::new(KvaFreeNode::new(kspace::TRACKED_MAPPED_PAGES_RANGE)));
+pub static KVA_FREELIST: SpinLock<BTreeMap<Vaddr, KvaFreeNode>> = SpinLock::new(BTreeMap::new());
```

Comment:
This variable does not need to be public. Remove the comments also.

This variable should also be defined in `impl Kva {}`. Use `Self::FREE_LIST` in the `Kva` implementations.